gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -563,6 +563,7 @@
    */
   public static final String TASK_STATE_COLLECTOR_INTERVAL_SECONDS = "task.state.collector.interval.secs";
   public static final int DEFAULT_TASK_STATE_COLLECTOR_INTERVAL_SECONDS = 60;
+  public static final String TASK_STATE_COLLECTOR_HANDLER_TYPE = "task.state.collector.handlerType";
```

Comment:
Change this to TASK_STATE_COLLECTOR_HANDLER or TASK_STATE_COLLECTOR_HANDLER_CLASS and allow plugging in the handler instead of restricting to the hard-coded enum options.